src/snowcli/connection_config.py

```diff
@@ -1,16 +1,16 @@
 from __future__ import annotations
 
-import configparser
 import os
 
+from configparser import ConfigParser
 
-class SnowsqlConfig:
+
+class ConnectionConfigs:
     def __init__(self, path="~/.snowsql/config"):
```

Comment:
I wonder if we shouldn't change also arg name from `path` to `snowsql_config_path`. I see it this way: 
* We have class ConnectionConfigs which holds configs of connections
* We create instances of this class using values from SnowSql config
* So we require snowsql config path in the constructor (not any path, but exactly snowsql config path)

IMO it would make the code more self-explanatory.